fix(active-index): resets index if activeIndex doesn't exist

NicholasLancey opened the pull request:
## Description

We discovered an edge case where the `activeIndex` did **not** exist. This often happens if the amount of `GalleryItems` changes based on the viewport width.

## Solution

Adds a check to see if the activeIndex exists, if it doesn't run the `goToIndex()` function and default to 0.

## Note

Fixes #3

andrewrubin commented on src/lib/components/gallery-main.jsx:
```diff
@@ -18,8 +18,13 @@ export const GalleryMain = ({ renderGalleryItem, className, ...props }) => {
   } = useGallery()
 
+  useEffect(() => {
+    if (!galleryItems[activeIndex]) goToIndex(0)
```
I think we can lose the `useEffect`, and integrate this logic into the actual `goToIndex` function. That thing should be updated to be a bit more defensive anyway. Inside that function, maybe instead of just accepting that `index` is defined, let's first check that it's valid, and if not, set to zero.

I think with this useEffect solution, we'll still briefly get a warning, since side-effects run _after_ the render phase.

marlonmarcello replied:
Agree with this one, otherwise we'll always have to resort to this `useEffect` trick.

NicholasLancey commented:
A bit of back and fourth with @andrewrubin on this one.

1. I've added a check to validate the index passed through to the `goToIndex` function. That's unrelated to the issue addressed in this PR.
2. I've moved the useEffect to the context. Unfortunately I can't think of another way for this to work - as the function needs to run every time `items` changes.

andrewrubin commented:
Agreed, thanks for entertaining the idea though. Tested on my end and working. Here's a screencap where `items` changes at 768:


https://github.com/wethegit/react-gallery/assets/30575213/03914509-dd93-4c26-a75c-7f27c8630ba3

andrewrubin commented:
Merging this one
